src/pkg/reverseproxy/reverseproxy.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reverseproxy
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	"github.com/goharbor/harbor/src/common/http/modifier"
+	"github.com/goharbor/harbor/src/common/utils/log"
+)
+
+// ReverseProxy defines the methods that a reverse proxy should implement
+type ReverseProxy interface {
+	ServeHTTP(rw http.ResponseWriter, req *http.Request)
+}
+
+// NewReverseProxy returns an instance of the reverse proxy
+func NewReverseProxy(url *url.URL, modifiers ...modifier.Modifier) ReverseProxy {
+	return &reverseProxy{
```

Comment:
refer to https://golang.org/src/net/http/httputil/reverseproxy.go
Can we reuse the semantics of director and modifer, where director modifies the request, modifer modifies the resposne?